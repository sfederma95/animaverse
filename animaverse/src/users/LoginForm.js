import React, {useState} from 'react';
import {useHistory} from 'react-router-dom';
import Alert from '../Alert'
import './login.css';

function LoginForm({login}){
    const history = useHistory();
    const [formData,setFormData]=useState({
        username: "",
        password: ""
    })
    const [formErrors,setFormErrors] = useState([])
    async function handleSubmit(e){
        e.preventDefault();
        let res = await login(formData);
        if (res.errors) {
            setFormErrors(res.errors)
        } else{
            history.push(`/users/${res.user.usr_id}`)
        }
    }
    function handleChange(e){
        const {name,value} = e.target;
        setFormData(l=>({...l,[name]:value}))
    }
    return(
        <div id='login-form'>
            <h1 id='tagline'>Login below</h1>
            <form id='form-el' onSubmit={handleSubmit}>
                <p id='login-p'>Login:</p>
                <div className='inputs'>
                    <label className = 'labels' htmlFor='username'>Username</label>
                    <input className='input-el' id='username' name='username' value={formData.username} onChange={handleChange} type='text' required/>
                </div>
                <div className='inputs'>
                    <label className = 'labels' htmlFor='password'>Password</label>
                    <input className='input-el' id='password' name='password' value={formData.password} onChange={handleChange} type='text' required/>
                </div>
                {formErrors.length ? <Alert messages={formErrors}/> : null}
                <button id='login-btn'>Login</button>
            </form>
        </div>
    )
}

export default LoginForm;