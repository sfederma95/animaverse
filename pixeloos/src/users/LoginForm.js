import React, {useState} from 'react';
import {useHistory} from 'react-router-dom';
import Alert from '../Alert'
import sleepCat from './sleepcat.svg';
import {hideCat, showCat} from './forms'

function LoginForm({login}){
    const history = useHistory();
    const [formData,setFormData]=useState({
        username: "",
        password: ""
    })
    const [formErrors,setFormErrors] = useState([])
    async function handleSubmit(e){
        e.preventDefault();
        let res = await login(formData);
        if (res.errors) {
            setFormErrors(res.errors)
        } else{
            history.push(`/user/${res.user.usr_id}`)
        }
    }
    function handleChange(e){
        const {name,value} = e.target;
        setFormData(l=>({...l,[name]:value}))
    }
    return(
        <div id='login-form'>
            <h1 id='tagline'>Login below</h1>
            <form id='form-el' onSubmit={handleSubmit}>
                <p id='login-p'>Login:</p>
                <div className='inputs'>
                    <img alt='sleep cat' className='sleep-cat' src={sleepCat}/>
                    <label className = 'labels' htmlFor='username'>Username</label>
                    <input onFocus = {showCat} onBlur = {hideCat} className='input-el good-input' id='username' name='username' value={formData.username} onChange={handleChange} type='text' required/>
                </div>
                <div className='inputs'>
                    <img alt='sleep cat' className='sleep-cat' src={sleepCat}/>
                    <label className = 'labels' htmlFor='password'>Password</label>
                    <input onFocus = {showCat} onBlur = {hideCat} className='input-el good-input' id='password' name='password' value={formData.password} onChange={handleChange} type='password' required/>
                </div>
                {formErrors.length ? <Alert messages={formErrors}/> : null}
                <button type='submit' id='login-btn'>Login</button>
            </form>
        </div>
    )
}

export default LoginForm;