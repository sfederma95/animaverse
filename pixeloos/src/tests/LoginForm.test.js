import React from 'react';
import {render} from '@testing-library/react';
import {MemoryRouter} from 'react-router';
import LoginForm from '../users/LoginForm';
import {UserProvider} from './testSetup';

it ('renders login form component without crashing', function(){
    render(
        <MemoryRouter>
            <UserProvider>
                <LoginForm />
            </UserProvider>
        </MemoryRouter>
    )
})

it ('matches snapshot', function(){
    const {asFragment} = render (
        <MemoryRouter>
            <UserProvider>
                <LoginForm/>
            </UserProvider>
        </MemoryRouter>
    )
    expect(asFragment()).toMatchSnapshot();
})