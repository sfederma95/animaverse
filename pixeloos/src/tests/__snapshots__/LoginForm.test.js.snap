// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    id="login-form"
  >
    <h1
      id="tagline"
    >
      Login below
    </h1>
    <form
      id="form-el"
    >
      <p
        id="login-p"
      >
        Login:
      </p>
      <div
        class="inputs"
      >
        <img
          alt="sleep cat"
          class="sleep-cat"
          src="sleepcat.svg"
        />
        <label
          class="labels"
          for="username"
        >
          Username
        </label>
        <input
          class="input-el good-input"
          id="username"
          name="username"
          required=""
          type="text"
          value=""
        />
      </div>
      <div
        class="inputs"
      >
        <img
          alt="sleep cat"
          class="sleep-cat"
          src="sleepcat.svg"
        />
        <label
          class="labels"
          for="password"
        >
          Password
        </label>
        <input
          class="input-el good-input"
          id="password"
          name="password"
          required=""
          type="text"
          value=""
        />
      </div>
      <button
        id="login-btn"
        type="submit"
      >
        Login
      </button>
    </form>
  </div>
</DocumentFragment>
`;
